citre-core.el

```diff
@@ -283,24 +283,33 @@ fields of info are updated."
 
 ;;;;; Info: path
 
-(defun citre-core--tags-file-use-relative-path-p (tagsfile)
-  "Detect if file paths in tags file TAGSFILE are relative.
-TAGSFILE is the path to the tags file.  This is done by
-inspecting the first line of regular tags."
-  (let* ((record (car (citre-core-get-records
-                       tagsfile nil nil nil
-                       :require '(input) :lines 1))))
-    (if (null record)
-        (error "Invalid tags file")
-      (not (file-name-absolute-p (citre-core-get-field 'input record))))))
-
+;; TODO: Test this on Windows.
 (defun citre-core--get-path-info (tagsfile)
   "Get path info of tags file TAGSFILE.
 See `citre-core--tags-file-info-alist' to know about the return
 value.  It is a valid value field of the `path' field."
-  (cons (citre-core--tags-file-use-relative-path-p tagsfile)
-        (nth 1 (car (citre-core-get-pseudo-tags
-                     "TAG_PROC_CWD" tagsfile)))))
+  (let* ((record (car (citre-core-get-records
+                       tagsfile nil nil nil
+                       :filter (not (citre-core-build-filter
+                                     'input "/" 'prefix))
+                       :require '(input) :lines 1)))
+         (cwd-ptag (nth 1 (car (citre-core-get-pseudo-tags
+                                "TAG_PROC_CWD" tagsfile))))
+         (cwd-guessed (file-name-directory tagsfile)))
+    (if (null record)
+        (cons nil cwd-ptag)
+      (cons t (or cwd-ptag
+                  ;; This is not 100% reliable, but there's little chance that
+                  ;; an irrelevant file happens to exist in the directory of
+                  ;; TAGSFILE, to which its relative path equals the input
+                  ;; field.
+                  (when (file-exists-p
+                         (expand-file-name (gethash 'input record)
+                                           cwd-guessed))
+                    cwd-guessed)
+                  (read-directory-name
+                   (format "Tags file root dir (= parent dir of %s): "
```

Comment:
@rrthomas I'd like to ask for your opinion on this prompt. What we do here is getting the first tag that uses relative path in the tags file, and let's assume the path is "some/file". What I want to ask for the user is:

"Tags file root dir, which is, the parent dir of some/file"

I have to show a shorter version since after all it's a prompt. It doesn't look very good, especially when the user is not familiar with the whole project, and doesn't know there's a `some/file` in the project.

Reply:
I would just ask the simpler question "Project root: (default %s)" where "%s" is the parent directory of the tags table. Or, if you know that is definitely the *wrong* answer, then simply don't offer a default, and if the user gives no answer, raise an error.

Reply:
I think this prompt is not very good. The concept of "project" does not appear in the core layer, so in the upper layers, we may offer things like "using different tags file for different parts of a project", and the root dir of the tags file may not be the project dir. Also, the project detection is handled automatically (see `citre-project-root`), so asking for project root may feel weird.

I'd also like to not bother the user if `/parent-dir-of-tags/relative-path-of-one-tag` exists, as the implementation now. Though there's a chance of mistake, I think it's solvable by upper components. I've thought about a `citre-doctor` command which shows you the project & tags file information, and if the user found the root path is wrong, we could advise they write the `TAG_PROC_CWD` pseudo tag, or move the file location & modify `citre-tags-files` accordingly.

How about changing this to "Root dir of tags file %s" where %s is the tags file path?

Reply:
> I think this prompt is not very good. The concept of "project" does not appear in the core layer

Agreed, it's important to know what the concept is. It's a bit unfortunate, though, if it is something that has little meaning to the user. I agree with your suggestion: if it is impossible to say clearly what "unit" of code we are looking for, then being concrete is the best remaining option.

> I'd also like to not bother the user if `/parent-dir-of-tags/relative-path-of-one-tag` exists, as the implementation now.

I agree: I assumed that you would do this!

> How about changing this to "Root dir of tags file %s" where %s is the tags file path?

Yes! (As above.)

Reply:
Ok, I'll merge this. Thanks for reviewing!